corehq/form_processor/interfaces/processor.py

```diff
@@ -256,18 +256,12 @@ def get_case_with_lock(self, case_id, lock=False, wrap=False):
         :return: tuple(case, lock). Either could be None
         :raises: IllegalCaseId
         """
-        # check across Couch & SQL to ensure global uniqueness
-        # check this domains DB first to support existing bad data
-        from corehq.apps.couch_sql_migration.progress import couch_sql_migration_in_progress
-
         case, lock = self.processor.get_case_with_lock(case_id, lock, wrap)
         if case:
             return case, lock
 
-        if not couch_sql_migration_in_progress(self.domain) and not settings.ENTERPRISE_MODE:
```

Comment:
what about this `ENTERPRISE_MODE` check, is that no longer relevant?

Reply:
My understanding is that that just skipped this extra check in enterprise envs where we knew there was no data in the "other" backend. I think it may have been an optimization for ICDS. Also, it seems unnecessary since all cases have been migrated out of Couch. @snopoke do agree?

Reply:
This will still check couch for every new case to make sure the ID doesn't exist. The optimisation was added for ICDS and probably not that necessary for other envs (depending on their scale). 

Since the migration is now complete I expect we'll be able to remove the dual processing code too in which case this will get removed then anyway.